### PR TITLE
key_length and confirmed emails

### PR DESCRIPTION
**Pull request by lehins**
So, first of all, I wanted to thank for a great app.
Couple things I thought could be useful to add
- ability to add confirmed emails in case they were confirmed in some other way (django-registration for instance)
- `key` field's `max_length` is set to 40, but there is no way to override length of generated key without  customizing the `EmailAddressManager`
  So this pull request addresses both issues.
  All tests pass, but no new tests was added.

### CODE REVIEW

**Comment by mfogel** [comment]
Thanks for the pull request.

I'm down to accept a customizable key length as a featue. I would like to see it set up such that the default key length is only specified one place - I believe in your pull request it's specified in both the model definition and in the manager.

As far as the ability to add confirmed emails... would

```
key = user.add_unconfirmed_email(email_addr)
user.confirm_email(key)
```

work for your use case?

**Comment by lehins** [comment]
@mfogel Thank you for your response.
After looking through my pull request I realized there is a much better way to structure it, for instance there is no need of having a `key_length` in adding an already confirmed email, hence I did some changes. Here are some arguments on design choice:
- I don't see any benefit in NOT adding a keyword argument `key_length` to `add_unconfirmed_email`, since it is merely a convenient shortcut to the manager's method `create_unconfirmed` with the only difference in return value.
- Adding unconfirmed email and then confirming it is a bad choice, I think, since it sends a signal `unconfirmed_email_created`, which is probably an undesired effect and it also would take two database hits instead of one. Plus it is nice to have a method `add_confirmed_email` for an app that deals only with email confirmation.
  So in short, couple things this pull request is about:
- ability to add confirmed emails, without any side effects
- ability to specify a variable length for confirmation key
- added tests for new features
- Also fixed a bug:
  - when called `EmailAddress.objects.create_unconfirmed(email, user=None)` was getting `AttributeError: 'EmailAddressManager' object has no attribute 'instance'`
    instead of intended `ValueError`, also added a test for this case.

**Comment by lehins** [comment]
@mfogel I noticed you did some more work on it, so I did merge it over to this pull request and fixed some conflicts. Tests are passing (travis gave an error because it had a problem installing django)

**Comment by mfogel** [comment]
Thanks for your work on this.

I've merged ea25d856917b48b3d7ca2199250142298bb538e6 into develop via the 'throw-correct-error' branch.

If you want to open up a new pull request that limits itself to just allowing the key_length to be specified on each applicable call, I'll consider it.

I appreciate the other miscellaneous fixes on this pull request but for clarity I'd rather see separate pull requests for each fix. I know it's more work... but I'm not going to merge a PR that doesn't have a narrow focus.

Thanks, closing this PR, please reopen a new one with focused scope.
